src/main/java/com/google/api/codegen/transformer/SurfaceNamer.java

```diff
@@ -117,10 +170,30 @@ public String getPageStreamingDescriptorName(Method method) {
     return varName(Name.upperCamel(method.getSimpleName(), "PageStreamingDescriptor"));
   }
 
+  public String getPageStreamingDescriptorConstName(Method method) {
+    return inittedConstantName(Name.upperCamel(method.getSimpleName()).join("page_str_desc"));
+  }
+
+  public String getBundlingDescriptorConstName(Method method) {
+    return inittedConstantName(Name.upperCamel(method.getSimpleName()).join("bundling_desc"));
+  }
+
   public void addPageStreamingDescriptorImports(ModelTypeTable typeTable) {
     // do nothing
   }
 
+  public void addBundlingDescriptorImports(ModelTypeTable typeTable) {
```

Comment:
Is it ever valid to construct an instance of SurfaceNamer, rather than a language-specific child class? If not, it could be marked abstract, and these methods could be made abstract.